### ISSUE TITLE
bigip_do resource fails incorrectly when provisioning apm

### ISSUE THREAD

**Issue by vinnie357** [F5Networks/tf-bigip-version0.12#61]
The DO endpoint responds with 404,500 while preforming provisioning tasks.
The resource errors in the task, however the provisioning completes successfully.

error:
```
bigip_do.do-apm: Creating...
bigip_do.do-apm: Still creating... [10s elapsed]
Error: Error while Sending/Posting http request with DO json :{"code":404,"message":"URI path /mgmt/shared/declarative-onboarding not registered.  Please verify URI is supported and wait for /available suffix to be responsive.","rest
OperationId":4232144,"kind":":resterrorresponse"}  <nil>
  on do.tf line 20, in resource "bigip_do" "do-apm":
  20: resource "bigip_do"  "do-apm" {
```

example bash handling the api responses:
https://github.com/vinnie357/f5-vpn-tf-cloud/blob/dev/terraform/vpn/templates/onboard_v2.tpl#L778

**Comment by papineni87** [comment]
I think resource is only looking for 200/202 http responses and gets failed for other responses. 
Will add logic to wait for 200 until the timeout ( default/ user defined ) and skip all other response codes within timeout.

**Comment by focrensh** [comment]
Resolved via TER-295 for release 1.2.
